src/Native/System.Net.Security.Native/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+project(System.Net.Security.Native)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+add_definitions(-DPIC=1)
+
+if (HAVE_HEIMNTLM_HEADERS)
+   find_library(LIB_HEIMNTLM NAMES heimntlm PATHS /usr/heimdal/lib)
+   include_directories(/usr/heimdal/include)
+
+   if(LIB_HEIMNTLM STREQUAL LIB_HEIMNTLM-NOTFOUND)
+       message(FATAL_ERROR "Cannot find libheimntlm.so and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+   endif()
+
+
+   set(NATIVENTLM_SOURCES
+     pal_ntlmapi.cpp
+   )
+
+   add_library(System.Net.Security.Native
+    SHARED
+    ${NATIVENTLM_SOURCES}
+   )
+
+   target_link_libraries(System.Net.Security.Native
+           ${LIB_HEIMNTLM}
+           ${OPENSSL_CRYPTO_LIBRARY}
+   )
+
+   install (TARGETS System.Net.Security.Native DESTINATION .)
+
+elseif (CMAKE_SYSTEM_NAME STREQUAL Linux)
+   message (FATAL_ERROR "cannot find libheim ntlm headers and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+else ()
```

Comment:
> If we didn't find the headers and we're Linux, fail.

I actually recommended against this.  The problem as I understand it is that the package one gets from the package managers produces some header conflicts with other important libraries, like libcurl, which means developers wanting to build this shim need to get and build the dependency from source.  Since this functionality exists purely for NegotiateStream which, while important, is niche, I suggested that for the time being, until we can come up with better ways to fix or workaround the conflict (or even change our dependency), we should not force every developer that wants to build corefx to first build this dependency from source, and rather only force those developers that need to build/use this shim locally to build the dependency from source.

Thoughts?

Assuming we're good with that approach, we'd have two options for how we handle the case of the missing headers:
1. Skip building the shim, resulting in runtime DllNotFoundExceptions when the developer then tries to use NegotiateStream.
2. Build the shim but with stub implementations that return failures, with the managed code detecting the failure and throwing an exception about the stub being built incorrectly.

Since we expect this to hopefully be a short-term thing, and for our build lab to always do the right thing such that actual users aren't affected, I'm thinking (1).